provider/aws/config.go

```diff
@@ -80,11 +94,25 @@ func CreateV2Configs(cfg *externaldns.Config) map[string]awsv2.Config {
 }
 
 func newV2Config(awsConfig AWSSessionConfig) (awsv2.Config, error) {
+	tlsConfig, err := tlsutils.NewTLSConfig(awsConfig.TLSCertPath, awsConfig.TLSCertKeyPath, awsConfig.TLSCAFilePath, "", awsConfig.TLSSkipVerify,
```

Comment:
I know, current implementation is to instantiate a tls config within the provider (ref: rfc2136,pdns,coredns), how about to consider an option to create tlsConfig outside the provider and only pass a pointer to *tls.Config ? What do you think?

Reply:
you mean like pass in as a variable to newV2Config? what do you suggest like put it before we do the switch case of providers (in main.go) and then any provider can use it if and when we implement it?

Reply:
Yes. This is a suggestion to review, as not sure where or not creating a tls cert within a provider is a scalable solution. No need to change that right now. More like something that worth to consider as an alternative, especially in light, that all providers should support this sort of configuration, not just AWS